docs/porting/security/psa_targets_in_mbed-os.md

```diff
@@ -0,0 +1,489 @@
+            "PSA"
+        ],
+        "extra_labels_add": [
+            "TFM",
+            "TFM_V8M"
+        ]
+```
+
+## Enabling PSA at application level
+
+Having an entropy source is crucial for Mbed TLS and Mbed Crypto. The [entropy source porting document](../porting/entropy-sources.html) discusses why and how to add an entropy source.
+
+If your target doesn't have a True Random Number Generator (TRNG), configure it as a non-PSA target in `targets/targets.json`. In this scenario, if an application wants to use the target as a PAS target, then it is the application's responsibility to provide an entropy source and mark that target as a PSA target at the application level. To enable PSA for a target from an application, use the config option [target_overrides](,,/reference/configuration.html).
+
+An example of `mbed_app.json`:
+
+```json
+"target_overrides": {
+    "K64F": {
+        "inherits": ["PSA_V7_M"]
+    }
+}
+```
+
+## Build and validation
+
+The Python script `build_tfm.py` automates building TF-M and copying the TF-M binary to the location defined by the target attribute `tfm_delivery_dir`. Another Python script, `build_psa_compliance.py`, automates
+building PSA API compliance tests.
+
+The [mbed-os-tf-m-regression-tests repository](https://github.com/ARMmbed/mbed-os-tf-m-regression-tests) contains build scripts, TF-M regression tests and PSA API compliance tests.
+
+### Building TF-M and running regression tests
+
+To build TF-M and regression tests:
+
+1. Clone the [mbed-os-tf-m-regression-tests](https://github.com/ARMmbed/mbed-os-tf-m-regression-tests) repository.
+1. Update `mbed-os.lib` to the copy of Mbed OS that contains new target support, including the `targets.json` changes described in this document.
+1. Run:
+
+    ```
+    mbed deploy
+    ```
+
+1. Run:
+
+    ```
+    python3 build_tfm.py -m <new target> -t <toolchain> -c ConfigRegressionIPC.cmake
+    ```
+
+    The command builds a TF-M binary the service necessary to run regression tests.
+
+    <span class="notes">**Note:** Use this `ConfigRegressionIPC.cmake` only to run regression tests. The secure binary copied to Mbed OS (when you are ready to merge your new target) must be generated using the config option `ConfigCoreIPC.cmake`.</span>
+
+1. Ensure the TF-M binary was copied to the location defined by the target attribute `tfm_delivery_dir`. If no binary was created or copied, please raise an [issue on the Mbed OS repository](https://github.com/ARMmbed/mbed-os).
+
+1. To build Mbed OS and run regression tests, run:
+
+    ```
+    mbed compile -m <new target> -t <toolchain>
+    ```
+1. Flash the regression tests binary to the target and ensure all regression tests pass.
+
+### Building TF-M and running PSA compliance tests
+
+To build TF-M and compliance tests:
+
+1. Switch to the `mbed-os-tf-m-regression-tests` directory.
+1. Edit `main.cpp` and set:
+
+    - `RUN_PSA_COMPLIANCE_TESTS` to `1`.
+    - `RUN_REGRESSION_TESTS` to `0`.
```

Comment:
This isn't how to select the tests anymore. We can use mbed_app.json.

Reply:
fixed